File: csharp/Minesweeper.Engine/Cell.cs
namespace MineSweeper.Engine
{
    public class Cell
    {
        public bool IsMine { get; set; } = false;
        public bool IsRevealed { get; set; } = false;
        public bool IsFlagged { get; set; } = false;
        public bool IsQuestioned { get; set; } = false;
        public bool Exploded { get; set; } = false;
        public bool WrongFlag { get; set; } = false;
        public int AdjacentMines { get; set; } = 0;
    }
}

File: kotlin/src/main/kotlin/engine/Cell.kt
package org.minesweeper.engine

data class Cell(
    var isMine: Boolean = false,
    var isRevealed: Boolean = false,
    var adjacentMines: Int = 0,
    var isFlagged: Boolean = false,
    var isQuestioned: Boolean = false,
    var exploded: Boolean = false,
    var wrongFlag: Boolean = false,
)
